pkg/kv/kvserver/replica_consistency.go

```diff
@@ -659,6 +672,11 @@ func (*Replica) sha512(
 				iter, span.Start, span.End, 0 /* nowNanos */, visitor,
 			)
 			iter.Close()
+			// Consume the remaining quota borrowed in the visitor. Do it even on
+			// iteration error, but prioritize returning the latter if it occurs.
+			if wErr := limiter.WaitN(ctx, batchSize); wErr != nil && err == nil {
```

Comment:
This is called multiple times (but only a few, as many as spans `MakeReplicatedKeyRangesExceptLockTable` returns), but I figured that it's ok to do a simple copy of this code block instead of trying to call it once.

Reply:
That's fine, it's only about four spans or so.